src/ngraph_encapsulate_op.cc

```diff
@@ -282,9 +289,39 @@ class NGraphEncapsulateOp : public OpKernel {
                         ng_function);
 #endif
       }
-
+      // Evict the cache if the number of elements exceeds 16
+      const char* cache_depth_specified =
+          std::getenv("NGRAPH_TF_FUNCTION_CACHE_ITEM_DEPTH");
+      if (cache_depth_specified != nullptr) {
+        NGRAPH_TF_FUNCTION_CACHE_ITEM_DEPTH = atoi(cache_depth_specified);
```

Comment:
`atoi` might be unsafe, for example it might not throw any errors when the user mischievously sets `NGRAPH_TF_FUNCTION_CACHE_ITEM_DEPTH=A_POTATO`

`stoi` might be slight safer, so we could convert the `char*` to a `string`.

However I am not exactly sure about `atoi` and `stoi`'s susceptibility to the `A_POTATO` attack. We might consider implementing a `is_valid_int` function to go through each char, and verify its a valid ASCII for an int. If this passes only then we call `atoi`

Reply:
We are reading `NGRAPH_TF_FUNCTION_CACHE_ITEM_DEPTH`everytime in the `Compute` function. We should move this to some place like backend manager, or atleast to the constructor of the encapsulate op.

However if the intention is that we are able to change the `NGRAPH_TF_FUNCTION_CACHE_ITEM_DEPTH` while the graph is running, then we can keep it here.

Reply:
Marking this resolved, as discussed